### PR TITLE
RavenDB-15917 Cached useOldFailureCheckOperation to static

### PR DESCRIPTION
**Pull request by haludi**
https://issues.hibernatingrhinos.com/issue/RavenDB-15917

### CODE REVIEW

**Comment by haludi** [comment]
The only way I can see that prevents us to cache the decision to use the old health check is if the health check fails for another reason then version mismatch.
As I understand it, when a request from a client fails for `RouteNotFoundException` and the client has a bigger version than the sever wrap the exception with `ClientVersionMismatchException`.
I can switch the health check failure catch condition so it will check **any** exception if contains `RouteNotFoundException` in its massage (now we check only if the exception type is `ClientVersionMismatchException`) just to be sure but I don't like it.

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public class RequestExecutor : IDisposable
 
         private static readonly GetStatisticsOperation BackwardCompatibilityFailureCheckOperation = new GetStatisticsOperation(debugTag: "failure=check");
         private static readonly DatabaseHealthCheckOperation FailureCheckOperation = new DatabaseHealthCheckOperation();
-        private ConcurrentSet<string> _useOldFailureCheckOperation;
+        private static ConcurrentSet<string> _useOldFailureCheckOperation;
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecutor.cs:57]: That isn't safe. What happens when we have two concurrent document stores to different servers?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Http/RequestExecutor.cs (hunk 1 above)
Comment (haludi) [reply src/Raven.Client/Http/RequestExecutor.cs:57]: Different server with different Urls so the only downside is a little bit bigger dictionary and so a little bit more time to add/search, no?

Comment (ayende) [reply src/Raven.Client/Http/RequestExecutor.cs:57]: The value of the `string` is the URL, node the node id?

Comment (haludi) [reply src/Raven.Client/Http/RequestExecutor.cs:57]: What do you mean by "node id"? The tag?
The [current implementation](https://github.com/ravendb/ravendb/pull/11607/files/8df03d1ae961740b66fe814e0eb725502bb9dd80#diff-8eb5dec408a858236268c82f3fe1523a8d957370e6a24bef4e21079f70d36ae1R1768) uses the URL includes the protocol (http/s).

Comment (ayende) [reply src/Raven.Client/Http/RequestExecutor.cs:57]: If the URL is there, fine by me

**Pull request merged**